src/batty.c

```diff
@@ -0,0 +1,430 @@
+#include "batty.h"
+#include "kirby.h"
+#include "functions.h"
+#include "object.h"
+#include "task.h"
+
+struct Unk_08354978 {
+    s16 unk0;
+    s16 unk2;
+    s16 unk4;
+    s16 unk6;
+    u8 unk8;
+    u8 unk9;
+}; 
```

Comment:
I believe this is just the existing `struct Unk_08353510`. Indexed and Used in the same way.